src/go/rdctl/cmd/api.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"os"
+	"regexp"
+)
+
+var apiSettings struct {
+	Method    string
+	InputFile string
+	Body      string
+}
+
+// apiCmd represents the api command
+var apiCmd = &cobra.Command{
+	Use:   "api",
+	Short: "Run API endpoints directly",
+	Long: `Runs API endpoints directly.
+Default method is PUT if a body or input file is specified, GET otherwise.
+
+Two ways of specifying a body:
+1. --input FILE: For example, '--input .../rancher-desktop/settings.json'. Specify '-' for standard input.
+
+2. --body|-b string: For the 'PUT /settings' endpoint, this must be a valid JSON string.
```

Comment:
Nitpicking now, but the body must **always** be a valid JSON string. So no need to reference a specific endpoint. And the requirements for `--input` and `--body` are the same.

I'm also wondering if `--body` is really necessary. It is just syntactic sugar for `echo "" | rdctl api --input -` or  `rdctl api --input <(echo "")`. If you want to specify individual settings you are probably going to use `--field key=value` instead anyways (I know, not yet implemented).

Reply:
Yes, this is just feedback discussion and should not hold up merging the PR.

The `rdctl api` is not intended for inexperienced users; they should use `rdctl set`.

`rdctl api` is mostly for RD developers ("us"), and for people using the REST API inside their integration, so they can easily experiment with API endpoints.